src/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+    dl
+    pthread
+    ${ZLIB_LIBRARIES}
+    ${xz_LIBRARIES}
+)
+
+target_compile_definitions(appimagetool
+    PRIVATE -D_FILE_OFFSET_BITS=64
+    PRIVATE -DVERSION_NUMBER="${GIT_VERSION}"
+    PRIVATE -DENABLE_BINRELOC
+)
+
+add_dependencies(appimagetool mksquashfs squashfuse)
+
+
+add_executable(appimaged appimaged.c notify.c elf.c getsection.c)
+
+target_link_libraries(appimaged
+    pthread
+    dl
+    ${CAIRO_LIBRARIES}
+    ${GLIB_LIBRARIES}
+    ${GIO_LIBRARIES}
+    ${inotify-tools_LIBRARIES}
+    ${squashfuse_LIBRARIES}
+    ${xz_LIBRARIES}
+    ${ZLIB_LIBRARIES}
+    ${libarchive_LIBRARIES}
+)
+
+target_compile_definitions(appimaged
+    PRIVATE -DVERSION_NUMBER="${GIT_VERSION}"
+    PRIVATE -D_FILE_OFFSET_BITS=64
+    PRIVATE -DHAVE_LIBARCHIVE3=0
+)
+
+add_dependencies(appimaged squashfuse)
+
+if(NOT USE_SYSTEM_INOTIFY_TOOLS)
+    add_dependencies(appimaged inotify-tools)
+endif()
+if(NOT USE_SYSTEM_LIBARCHIVE)
+    add_dependencies(appimaged libarchive)
+endif()
+
+add_executable(AppRun AppRun.c)
+
+
+add_executable(validate validate.c getsection.c)
+
+target_link_libraries(validate
+    ${GLIB_LIBRARIES}
+    ${OPENSSL_LIBRARIES}
+    )
+
+
+add_executable(digest digest.c getsection.c)
+
+target_link_libraries(digest
+    ${OPENSSL_LIBRARIES}
+    )
+
+
+# install binaries
+install(
+    PROGRAMS ${mksquashfs_INSTALL_DIR}/mksquashfs ${CMAKE_CURRENT_BINARY_DIR}/runtime
+    DESTINATION bin
+)
+install(
+    TARGETS AppRun appimaged appimagetool digest validate
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib/static
+    INCLUDES DESTINATION include
+)
+
+
+# install libappimage
+install(TARGETS libappimage
+    EXPORT AppImageKit
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib/static
+    PUBLIC_HEADER DESTINATION include
+)
+
+# Add all targets to the build-tree export set
+export(
+    TARGETS libappimage
+    FILE "${PROJECT_BINARY_DIR}/AppImageTargets.cmake"
+)
+
+# Export the package for use from the build-tree
+# (this registers the build-tree with a global CMake-registry)
+export(PACKAGE AppImageKit)
+
+# Create the AppImageConfig.cmake and AppImageConfigVersion files
+configure_file(
+    "${PROJECT_SOURCE_DIR}/cmake/AppImageKitConfig.cmake.in"
+    "${PROJECT_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/AppImageKitConfig.cmake"
+    @ONLY
+)
+# ... for both
+configure_file(
+    "${PROJECT_SOURCE_DIR}/cmake/AppImageKitConfigVersion.cmake.in"
+    "${PROJECT_BINARY_DIR}/AppImageKitConfigVersion.cmake"
+    @ONLY
+)
+
+# Install the AppImageConfig.cmake and AppImageConfigVersion.cmake
+install(FILES
+    "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/AppImageKitConfig.cmake"
+    "${PROJECT_BINARY_DIR}/AppImageKitConfigVersion.cmake"
+    DESTINATION "/lib/CMake/AppImage"
+    COMPONENT dev
+)
+
+# Install the export set for use with the install-tree
+install(
+    EXPORT AppImageKit
+    DESTINATION "/lib/CMake/AppImage"
```

Comment:
cmake modules path changes from one distribution to another it should be somehow configurable